src/device.rs

```diff
@@ -2297,12 +2295,8 @@ impl TaskRuntime for Runtime {
                                 // We have downgraded the connection. Notify cross ping check about that.
                                 direct_entities.cross_ping_check.notify_failed_wg_connection(public_key)
                                     .await?;
-
                                 direct_entities.session_keeper.remove_node(&public_key).await?;
-
-                                let mut event = AnalyticsEvent::from_event(&mesh_event);
-                                event.peer_state = PeerState::Connected;
-                                self.entities.aggregator.change_peer_state_relayed(&event).await;
+                                direct_entities.upgrade_sync.clear_accepted_session(public_key).await;
```

Comment:
We only clear session on downgrade? Should we not also clear it on connection to direct, or is that not detectable currently?